### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by mira1zu**

### CODE REVIEW

**Review by sancho325 — APPROVED** [review]

File: src/robotFactory.js [hunk 1]
```diff
@@ -1,10 +1,95 @@
...
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  getInfo() {
+    return `Robot: ${this.name}, `
+          + `Chip version: ${this.chipVersion}, `
+          + `Weight: ${this.weight}`;
```
Comment (illiarerun) [inline src/robotFactory.js:39]: remember about proper indents
```suggestion
    return `Robot: ${this.name}, `
      + `Chip version: ${this.chipVersion}, `
      + `Weight: ${this.weight}`;
```

File: src/robotFactory.js [hunk 2]
```diff
@@ -1,10 +1,95 @@
...
+  constructor(name, weight, coords = {}, chipVersion) {
+    const defaultCoords = {
+      x: 0,
+      y: 0,
+    };
 
-class FlyingRobot {}
+    const robotCoords = Object.keys(coords).length !== 2
+      ? defaultCoords
+      : coords;
```
Comment (illiarerun) [inline src/robotFactory.js:12]: but what if `coords` obj has some different keys?

File: src/robotFactory.js [hunk 3]
```diff
@@ -1,10 +1,95 @@
...
+
+    const defaultCoords = {
+      x: 0,
+      y: 0,
+      z: 0,
+    };
+
+    const robotCoords = Object.keys(coords).length !== 3
+      ? defaultCoords
+      : coords;
```
Comment (illiarerun) [inline src/robotFactory.js:55]: you repeat logic of parent constructor here, try to do it a bit other way, all you need is to append `z` coord in this constructor

File: src/robotFactory.js [hunk 4]
```diff
@@ -1,10 +1,95 @@
...
+  }
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(name,
+    weight,
+    coords = {},
+    chipVersion,
+    maxLoadWeight,
+    currentLoad) {
```
Comment (illiarerun) [inline src/robotFactory.js:75]: proper formatting would look like this
```suggestion
  constructor(
    name,
    weight,
    coords = {},
    chipVersion,
    maxLoadWeight,
    currentLoad,
  ) {
```

File: src/robotFactory.js [hunk 5]
```diff
@@ -1,10 +1,95 @@
...
+    super(name, weight, coords, chipVersion);
+
+    this.maxLoadWeight = maxLoadWeight;
+    this.currentLoad = currentLoad;
+  }
+
+  hookLoad(cargo) {
+    if ((this.currentLoad === null)
+      && (cargo.weight <= this.maxLoadWeight)) {
```
Comment (illiarerun) [inline src/robotFactory.js:84]: I believe you can fit it in one line
```suggestion
    if (this.currentLoad === null && cargo.weight <= this.maxLoadWeight) {
```

File: src/robotFactory.js [hunk 6]
```diff
@@ -1,10 +1,95 @@
...
+  constructor(name,
+    weight,
+    coords = {},
+    chipVersion,
+    maxLoadWeight,
+    currentLoad) {
+    super(name, weight, coords, chipVersion);
+
+    this.maxLoadWeight = maxLoadWeight;
+    this.currentLoad = currentLoad;
```
Comment (illiarerun) [inline src/robotFactory.js:79]: what if `currentLoad` is undefined? your check in `hookLoad` won't work properly then

**Review by illiarerun — CHANGES_REQUESTED** [review]
GJ, now let's work on some fixes:

File: src/robotFactory.js [hunk 7]
```diff
@@ -1,10 +1,81 @@
...
+    currentLoad = null
+  ) {
+    super(name, weight, coords, chipVersion);
+
+    this.maxLoadWeight = maxLoadWeight;
+    this.currentLoad = currentLoad;
+  }
+
+  hookLoad(cargo) {
+    if ((this.currentLoad === null) && (cargo.weight <= this.maxLoadWeight)) {
```
Comment (mykola-koval) [inline src/robotFactory.js:70]:
```suggestion
    if (!this.currentLoad && cargo.weight <= this.maxLoadWeight) {
```

**Review by mykola-koval — APPROVED** [review]

**Pull request closed without merging**